lib/document.js

```diff
@@ -624,6 +651,65 @@ class Document extends Readable {
     this._begin(null)
     this._pending.push(() => external.setAsTemplate(this))
   }
+
+  outline(title, destination, parent) {
```

Comment:
Implementing the outline method here, would restrict its usage to the root document object, i.e., it would not be possible to create an outline element within for example a table cell. We could consider moving it to the `Fragment` class (which Document and others inherit from; `lib/fragment.js`).

Reply:
I moved the method into the Fragment object and assigned it to the document object accordingly. I also updated the types.

However, I'm not sure that it is relevant for the outline spec itself.

Since an outline is not part of the PDF document's body (it cannot be printed, for example), to me it only makes sense to add it with `doc.outline()`. I would find confusing to call `table.outline()`, since the outline is not tied to the table itself, but points to a destination inside of the table.

In comparison, I understand better how the same argument applies for the destination that the outline points to, and that's how you implemented it.

But in that case, a level of abstraction doesn't do any harm and may save some refactoring in the future, I guess.

Reply:
You have a good point, I'll go and merge it and do a slight adjustment to the code. Afterwards I will give an example why we might have to call outline from a table cell.

Thanks again for this PR!